types/module/module.go

```diff
@@ -92,29 +94,55 @@ func (bm BasicManager) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 }
 
 // RegisterInterfaces registers all module interface types
-func (bm BasicManager) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+func (bm BasicManager) RegisterInterfaces(registry types.InterfaceRegistry) {
 	for _, m := range bm {
 		m.RegisterInterfaces(registry)
 	}
 }
 
 // DefaultGenesis provides default genesis information for all modules
 func (bm BasicManager) DefaultGenesis(cdc codec.JSONCodec) map[string]json.RawMessage {
-	genesis := make(map[string]json.RawMessage)
+	genesisData := make(map[string]json.RawMessage)
 	for _, b := range bm {
-		if mod, ok := b.(HasGenesisBasics); ok {
-			genesis[b.Name()] = mod.DefaultGenesis(cdc)
+		// first check if the module is an adapted Core API Module
+		if mod, ok := b.(coreAppModuleBasicAdapator); ok {
```

Comment:
Maybe this is okay... I wonder if the adaptor should just implement the legacy default genesis and validate genesis methods instead..:

Reply:
What about implementing all methods from AppModule, AppModuleBasic, etc. in the adapter? Would that be an issue?

Reply:
I don't think it would be an issue, but it might be enough to just implement AppModuleBasic + the genesis methods

Reply:
ok, just implemented HasGenesis in coreAppModuleBasicAdapator and removed all checks for that type, given that:

```go
var _ AppModuleBasic = coreAppModuleBasicAdapator{}
var _ HasGenesis = coreAppModuleBasicAdapator{}
```

BTW, we can't return validator updates right?

Reply:
Ah ok, that's gonna be `ValidatorUpdateService`. We'll be adding that later right?